modules/core/src/v2/coins/eos.ts

```diff
@@ -813,7 +858,26 @@ export class Eos extends BaseCoin {
       }
 
       const transactionHeaders = yield self.getTransactionHeadersFromNode();
-      const eosClient = new EosJs({ chainId: self.getChainId(), transactionHeaders });
+      if (!transactionHeaders) {
+        throw new Error('Could not get transaction headers from node');
+      }
+      const headers: EosTransactionHeaders = transactionHeaders;
+
+      // drop milliseconds and trailing Z from expiration
+      const date = new Date(headers.expiration as string);
```

Comment:
please be very defensive when it comes to date manipulations and I'd recommend even using the `moment` library for this.

The behavior of `Date.prototype.toString` differs between nodejs and browsers:

browser:
```
new Date()
Thu Oct 14 2021 14:08:10 GMT-0700 (Pacific Daylight Time)
```

node:
```
> new Date()
2021-10-14T21:10:09.938Z
```